### PR TITLE
feat: added check for valid git commit and warning message

### PR DESCRIPTION
**Pull request by sahil9001**
### Description:
This PR addresses this issue #3394 , wherein we check the validity of git commit and throw the invalid commit message if commit doesn't match the GitHub SHA standard.

Screenshot 2024-10-15 at 8 56 34 PM

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -312,6 +312,18 @@ func main() {
 	}
 }
 
+func isValidCommit(commit string) bool {
+	if len(commit) != 40 {
+		return false
+	}
```
Comment (rgmz) [inline main.go:318]: I don't believe it's necessary to provide the full SHA. 

> Git is smart enough to figure out what commit you’re referring to if you provide the first few characters of the SHA-1 hash, as long as that partial hash is at least four characters long and unambiguous; that is, no other object in the object database can have a hash that begins with the same prefix.

https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection#_short_sha_1

Comment (sahil9001) [reply main.go]: Can you check now, I have fixed the issue, also how to link the issue with the PR? I can't see the option to do so

Comment (sahil9001) [reply main.go]: @rgmz

Comment (rgmz) [reply main.go]: Doing basic validation is helpful.

However, I think the only way to properly satisfy #3394 is to dispatch a command to Git. (e.g., https://stackoverflow.com/questions/18515488/how-to-check-if-the-commit-exists-in-a-git-repository-by-its-sha-1)

Comment (sahil9001) [reply main.go]: Yeah, you are right, I have fixed by dispatching this command from the function.

**Comment by ahrav** [comment]
Thanks @sahil9001, sorry for the delayed review here.

**Pull request merged**